README.md

```diff
@@ -92,3 +92,68 @@ Options:
   </body>
 </html>
 ```
+
+------------------------
+
+## Forking the Live Network
+
+The script `run-moonbeam-fork.ts` has been provided which allows you to fork the live state of the network and run them locally. 
+
+### Usage
+
+The simplest way to run a forked-network of `Moonbeam` is by calling:
+
+```
+npm run fork
+```
+
+Which will grab the latest polkadot and moonbeam binaries, grab the latest snapshot of live state, modify some values (such as validator keys, and inject balances),
+ and run it as a new local network. From here you can perform any interaction as if it was the real state, such as runtime upgrades, interactions on deployed contracts,
+ staking operations etc.
+
+
+If however, you are more interested in intereacting with the contracts in the Moonbeam emulated EVM environment, but not so much parachain staking or XCM, you can run
+the forked network as a solo node in dev mode. This allows for manual sealing of blocks which dramatically reduces execution times of tests (reduction of 12s blocktime into milliseconds).
+
+```
+npm run fork-solo
+```
+
+### Further Examples
+
+Calling the script directly can be done via: `npx ts-node ./src/tools/run-moonbeam-fork.ts`
+
+The minimal command is: `npx ts-node ./src/tools/run-moonbeam-fork.ts -n moonbeam`
+
+By default the polkadot and moonbeam binaries will be downloaded from github if none found in the `binaries` folder, but if for whatever reason you need to provide your own (e.g. you are on an Apple Silicon chip)
+use the `--polkadot-binary` option to provide the path to the binary to use (or copy them into the folder). If doing this option, make sure the correct binary version is supplied via `--moonbeam-version` and 
+`--polkadot-version` respectively, otherwise the script will still attempt to download the latest.
+
+
+The full list of options can be found in the `--help` :
+```
+      --help                      Show help                            [boolean]
+      --version                   Show version number                  [boolean]
+  -n, --network                   Network to retrieve the exported state for
+                                                             [string] [required]
+      --latest                    Will verify if a latest snapshot is available
+                                  and download it     [boolean] [default: false]
+      --purge                     Will delete previous execution database
```

Comment:
Does this purge the specs too (I see there is a --purge-specs and I don't know which ones I would have to run if I want to fully clean it)

Reply:
purge and purge-specs were pre-existing options.

I've now renamed them to be `purge-all` and `reset-to-genesis` to be more clear about what each option does. setting the former you should use to completely reset your dir, the latter option you should use if you just want to reset your local state and skip having to re-import everything again